overlord/state/taskrunner.go

```diff
@@ -472,13 +513,15 @@ ConsiderTasks:
 		r.run(t)
 
 		running = append(running, t)
+
+		// If a change had jobs ready to run, reset its exhaustion status.
+		r.exhausted[t.Change().ID()] = false
```

Comment:
as it's a bit annoying that we do not remove things from it once added

Reply:
Thanks to your comment the cache is now removed!